### ISSUE TITLE
Wunderground plugin crashing Homebridge

### ISSUE THREAD

**Issue by Tiger519** [xfjx/homebridge-wunderground#19]
I've started running into an issue where the Wunderground plugin receives some sort of error, which causes all of Homebridge to crash. I've mitigated the issue by running it in its own Homebridge instance, but I'm hoping for more reliability in order to use the temperature as a trigger for automations. Below is the error message I'm receiving (running on RPi Zero). Had only had a few issues here and there for the life of the plugin, but lately have been getting this almost every day.

Error fetching weather data from wunderground.com! Check your configuration.
/opt/node/lib/node_modules/homebridge-wunderground/index.js:82
                if (response['response']['error'] && response['response']['error']['type'] && response['response']['error']['description'])
                                        ^

TypeError: Cannot read property 'error' of undefined
    at Wunderground.<anonymous> (/opt/node/lib/node_modules/homebridge-wunderground/index.js:82:41)
    at Request._callback (/opt/node/lib/node_modules/homebridge-wunderground/node_modules/wundergroundnode/lib/wundergroundnode.js:148:22)
    at self.callback (/opt/node/lib/node_modules/homebridge-wunderground/node_modules/request/request.js:186:22)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at Request.onRequestError (/opt/node/lib/node_modules/homebridge-wunderground/node_modules/request/request.js:878:8)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at Socket.socketErrorListener (_http_client.js:309:9)
    at emitOne (events.js:96:13)

**Comment by jmissig** [comment]
I'm seeing the same.

**Comment by Tiger519** [comment]
@jmissig Weather Underground API has been down a bunch yesterday and today. If you try to hit the API URL directly, you’ll get an HTTP 503 error, or some sort of other failure to load. This plugin seems to have trouble when that happens and just shuts down. Not sure if there is a way to fix it. For me ideally the temp/humidity should be returned as null values (since the previous value won’t be correct) in this case, but sadly I don’t know enough about this stuff to figure out how to identify and handle a failure to get the json in the response and then set a default (e.g. null) value for the sensors.

**Comment by jmissig** [comment]
@Tiger519 yeah, I think the issue is line 82 doesn't check that `response['response']` exists before asking for its error. But looking at the source of wundergroundnode, any time the request returns anything other than status 200 (OK), that argument is not necessarily the expected response object.

I don't have nearly enough set up to try fixing + patching this myself though.

**Comment by Tiger519** [comment]
@jmissig Well, looking again at the code, it looks like that else statement is being used only for error logging. Regardless, it will log "Error fetching weather data from wunderground.com! Check your configuration.", the other lines just seem to be to try to log the error type and description, and the err returned by the response. I might try just commenting out 82/83 and see what happens.

**Comment by xfjx** [comment]
please try this code

`               if (response && resonse['response'] && response['response']['error'] && response['response']['error']['type'] && response['response']['error']['description'])
                    that.log(response['response']['error']['type'] + " : " + response['response']['error']['description']);
                if (err)
                    that.log(err);
                callback(that.weather);
`

**Comment by Tiger519** [comment]
@xfjx I finally encountered my first error fetching data since I updated the code. I simply commented out the lines trying to log the error type and description and it did indeed prevent the plugin from crashing Homebridge. It seems every time I was facing an error, the response didn't exist, so I think commenting out those lines would have the same effect as the code above. Not sure if that would be applicable for all users, though. For me, I think I'll stick with my modification since it's working and I'm always wary of making changes to something that's working!
